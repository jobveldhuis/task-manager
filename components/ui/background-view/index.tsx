import { ReactNode } from "react";
import { LinearGradient } from "expo-linear-gradient";
import { StyleSheet } from "react-native";
import Colors from "../../../constants/Colors";

type BackgroundViewProps = {
  children: ReactNode;
};

export function BackgroundView({ children }: BackgroundViewProps): JSX.Element {
  return (
    <LinearGradient
      style={styles.container}
      colors={[
        Colors.backgroundGradient.primary,
        Colors.backgroundGradient.secondary,
      ]}
      testID="background"
    >
      {children}
    </LinearGradient>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "flex-start",
    justifyContent: "flex-start",
    paddingHorizontal: 25,
    paddingVertical: 150,
  },
});
