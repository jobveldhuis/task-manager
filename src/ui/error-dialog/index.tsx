export { ErrorDialog } from "./error-dialog.component";
