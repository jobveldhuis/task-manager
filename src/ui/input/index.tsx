export { Input } from "./input.component";
export { DateInput } from "./date-input.component";
export { MultilineInput } from "./multiline-input.component";
