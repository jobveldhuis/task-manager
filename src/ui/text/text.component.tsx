import { Text as NativeText, StyleSheet, TextStyle } from "react-native";
import { COLORS } from "@/util/colors.const";

type TextProps = {
  children: string;
  variant?: (typeof variants)[number];
  style?: AllowedTextStyle;
};

export const variants = [
  "paragraph",
  "strikethrough",
  "button",
  "link",
  "menu",
  "menuSelected",
  "xl",
  "light",
] as const;

type AllowedTextStyle = Pick<TextStyle, "marginBottom">;

export function Text({
  children,
  style,
  variant = "paragraph",
}: TextProps): JSX.Element {
  return (
    <NativeText style={[styles.default, styles[variant], style]}>
      {children}
    </NativeText>
  );
}

const styles = StyleSheet.create({
  default: {
    fontSize: 16,
    fontWeight: "600",
  },
  paragraph: {
    lineHeight: 22,
    color: COLORS.secondary,
  },
  button: {
    color: COLORS.button.inverse,
    textAlign: "center",
    fontWeight: "bold",
  },
  link: {
    color: COLORS.secondary,
    fontWeight: "bold",
  },
  menu: {
    color: COLORS.inverse,
    fontSize: 16,
    fontWeight: "500",
  },
  menuSelected: {
    color: COLORS.inverse,
    fontSize: 20,
    fontWeight: "500",
  },
  xl: {
    fontSize: 42,
    fontWeight: "normal",
  },
  light: {
    fontWeight: "normal",
  },
  strikethrough: {
    textDecorationLine: "line-through",
    textDecorationStyle: "solid",
  },
});
