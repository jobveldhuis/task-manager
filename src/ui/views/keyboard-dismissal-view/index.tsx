export * from "./keyboard-dismissal-view.component";
