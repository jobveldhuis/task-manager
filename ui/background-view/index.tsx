import { ReactNode } from "react";
import { LinearGradient } from "expo-linear-gradient";
import { StyleSheet } from "react-native";
import { COLORS } from "@/util/colors.const";

type BackgroundViewProps = {
  children: ReactNode;
};

export function BackgroundView({ children }: BackgroundViewProps): JSX.Element {
  return (
    <LinearGradient
      style={styles.container}
      colors={[
        COLORS.backgroundGradient.primary,
        COLORS.backgroundGradient.secondary,
      ]}
      testID="background"
    >
      {children}
    </LinearGradient>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "flex-start",
    justifyContent: "flex-start",
    paddingHorizontal: 32,
    paddingVertical: 120,
  },
});
