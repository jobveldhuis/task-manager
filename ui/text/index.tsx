import { Text as NativeText, StyleSheet, TextStyle } from "react-native";
import { ReactNode } from "react";
import { COLORS } from "@/util/colors.const";

type TextProps = {
  children: ReactNode;
  variant?: (typeof variants)[number];
  style?: AllowedTextStyle;
};

export const variants = [
  "title",
  "inlineTitle",
  "paragraph",
  "blackParagraph",
  "button",
] as const;

type AllowedTextStyle = Pick<TextStyle, "marginBottom">;

export function Text({
  children,
  style,
  variant = "paragraph",
}: TextProps): JSX.Element {
  return (
    <NativeText style={[styles.default, styles[variant], style]}>
      {children}
    </NativeText>
  );
}

const styles = StyleSheet.create({
  default: {
    fontSize: 16,
  },
  title: {
    fontSize: 24,
    fontWeight: "bold",
  },
  inlineTitle: {
    fontWeight: "bold",
  },
  paragraph: {
    lineHeight: 22,
    color: COLORS.primary,
  },
  blackParagraph: {
    lineHeight: 22,
    color: COLORS.secondary,
  },
  button: {
    color: COLORS.button.inverse,
    textAlign: "center",
    fontWeight: "bold",
  },
});
