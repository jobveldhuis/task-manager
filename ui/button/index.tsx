import {
  GestureResponderEvent,
  Pressable,
  StyleSheet,
  View,
  ViewStyle,
} from "react-native";
import { COLORS } from "@/util/colors.const";
import { Text } from "../text";

type ButtonProps = {
  title: string;
  onPress?: (event: GestureResponderEvent) => void;
  style?: ViewStyle;
};

export function Button({
  title,
  style,
  onPress: handlePress,
}: ButtonProps): JSX.Element {
  return (
    <Pressable
      style={({ pressed: isPressed }) => [
        styles.container,
        { opacity: isPressed ? 0.6 : 1 },
      ]}
      onPress={handlePress}
    >
      <View style={[styles.button, style]}>
        <Text variant="button">{title}</Text>
      </View>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    width: "100%",
  },
  button: {
    backgroundColor: COLORS.button.primary,
    padding: 10,
    borderRadius: 20,
    width: "100%",
    textAlign: "center",
  },
});
