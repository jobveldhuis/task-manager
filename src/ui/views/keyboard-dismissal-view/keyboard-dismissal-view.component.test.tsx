import { render } from "@testing-library/react-native";
import { Text } from "@/ui/text";
import { KeyboardDismissalView } from "./keyboard-dismissal-view.component";

describe("KeyboardDismissalView", () => {
  const component = (
    <KeyboardDismissalView>
      <Text>Test</Text>
    </KeyboardDismissalView>
  );

  it("should match snapshot", () => {
    expect(render(component).toJSON()).toMatchSnapshot();
  });
});
