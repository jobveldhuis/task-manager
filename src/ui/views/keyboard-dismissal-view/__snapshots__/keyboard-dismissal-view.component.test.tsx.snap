// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardDismissalView should match snapshot 1`] = `
<Text
  style={
    [
      {
        "color": "#000C49",
        "fontSize": 16,
        "fontWeight": "600",
      },
      {
        "lineHeight": 22,
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;
